papermill/execute.py

```diff
@@ -195,11 +195,15 @@ def _parameterize_notebook(nb, kernel_name, parameters):
 
     # Remove the old cell and replace it with a new one containing parameter content.
     param_cell_index = _find_parameters_index(nb)
-    old_parameters = nb.cells[param_cell_index]
-    before = nb.cells[:param_cell_index]
+    if param_cell_index >= 0:
+        old_parameters = nb.cells[param_cell_index]
+    else:
+        old_parameters = nbformat.v4.new_code_cell() # Fake cell
+        old_parameters.metadata['tags'] = ['parameters']
+    before = nb.cells[:param_cell_index + 1]
```

Comment:
The current file on master has `before = nb.cells[:param_cell_index]`. Why are you adding `+1` to the slice? Wouldn't this include the param cell as well?

Reply:
Yes it does, that's the intention now. Insert instead of replace. This the cell you tagged is treated as prefix defaults to input parameters

Reply:
Input
![input](https://user-images.githubusercontent.com/1863892/34180340-2b5b0556-e4c3-11e7-945e-38482ad557cb.png)
Output
![output](https://user-images.githubusercontent.com/1863892/34180341-2d9f7e50-e4c3-11e7-890b-031879939ad9.png)

Reply:
got it

Reply:
This is great.  Do you think it makes sense to update the tag(s) of the output cells to reflect that one is the original cell and one is the newly appended cell with the passed parameters?

Reply:
@kainoa21 That sounds like a good idea for readability, do you want to make an issue for it as a reminder to make that improvement?

Reply:
https://github.com/nteract/papermill/issues/84